src/Common/src/TypeSystem/CodeGen/TypeDesc.CodeGen.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Internal.TypeSystem
+{
+    // Extensions to TypeDesc suitable for code generation purposes
+    partial class TypeDesc
+    {
+        /// <summary>
+        /// Validates that a type can be fully loaded. Throws an exception if a failure occurs.
+        /// </summary>
+        public virtual void ValidateCanLoad()
```

Comment:
I do not think that this validation should be on TypeDesc. This validation should be on EETypeNode instead. The question that it really needs to answer is whether we are able to construct complete EETypeNode in the image.